### PR TITLE
Fix execution layer redundancy

### PR DESCRIPTION
**Pull request by iamit-singh**
## Issue Addressed

#3307

## Proposed Changes

renamed and changed types(from Vec to Option) of below fields of Config struct:
execution_endpoints: Vec => execution_endpoint: Option
secret_files: Vec => secret_file: Option

### CODE REVIEW

File: beacon_node/beacon_chain/src/test_utils.rs [hunk 1]
```diff
@@ -408,21 +408,20 @@ where
...
 
-        let urls: Vec<SensitiveUrl> = urls
-            .iter()
-            .map(|s| SensitiveUrl::parse(s))
-            .collect::<Result<_, _>>()
-            .unwrap();
+        let url = match SensitiveUrl::parse(url) {
+            Ok(s) => Some(s),
+            Err(_) => None,
+        };
```
Comment (realbigsean) [inline beacon_node/beacon_chain/src/test_utils.rs]: you could add a bit of sugar here:
```suggestion
        let url = SensitiveUrl::parse(url).ok();
```

**Review by realbigsean — DISMISSED** [review]
Thanks for the contribution! It's a nice cleanup, just had one small suggestion

**Comment by realbigsean** [comment]
Looks like some tests need updating as well, if you want to check locally whether tests compile `make lint` will also check them

**Comment by iamit-singh** [comment]
@realbigsean thanks for review. Actually I ran `make lint` in my local, strangely it doesn't give any error. Anyways, I see the errors in here, I'll work on them.

**Comment by realbigsean** [comment]
> @realbigsean thanks for review. Actually I ran `make lint` in my local, strangely it doesn't give any error. Anyways, I see the errors in here, I'll work on them.

ah there must be some tests are are missed. usually i also run `make test-beacon-chain` and that seems to cover the rest

**Comment by iamit-singh** [comment]
@realbigsean fixed the tests!

**Review by realbigsean — APPROVED** [review]
Awesome!

**Comment by realbigsean** [comment]
@mergify queue

**Comment by realbigsean** [comment]
@mergify requeue

**Comment by michaelsproul** [comment]
@mergifyio requeue

**Pull request merged**